packages/core/lib/commands/migrate/determineDryRunSettings.js

```diff
@@ -1,19 +1,9 @@
-module.exports = function (config, options) {
-  // Source: ethereum.stackexchange.com/questions/17051
-  const networkWhitelist = [
-    1, // Mainnet (ETH & ETC)
-    2, // Morden (ETC)
-    3, // Ropsten
-    4, // Rinkeby
-    5, // Goerli
-    8, // Ubiq
-    42, // Kovan (Parity)
-    77, // Sokol
-    99, // Core
+const Ganache = require("ganache");
 
-    7762959, // Musiccoin
-    61717561 // Aquachain
-  ];
+module.exports = function (config, options) {
+  //note: this is a list of chain IDs but we're still using
+  //network ID.  This should be fixed later.
+  const supportedChainIds = Ganache.__experimental_info().fork.knownChainIds;
```

Comment:
`Ganache.__experimental_info()` implies the API behavior could change or disappear.

Reply:
Well it's not like this is an external package. Talk to the Ganache team and see what the plans are.

Reply:
Yeah, this is the name @davidmurdoch & co settled on; note that this feature was added to Ganache, because I asked for it, specifically for it to be used for this *exact* purpose!  So I intend to use it regardless of what David and company call it!

Reply:
The PR mentions the API will likely change in the future. 

> This introduces a new experimental and private (this will likely change in a future release!) __experimental_info export:

My concern is the risk of a Ganache patch release changing the behavior. I'm unsure how SemVer defines version bumps regarding experimental API changes.

Reply:
So, are you suggesting that we pin Ganache alongside making this change, then?  Or what?

Reply:
@gnidan, thoughts?

Reply:
Since this might break semver, I can see the argument for why we might want to pin.

But I'd still rather get that name switched before this PR gets merged, and then lean on semver (and expectations provided by Ganache) to stick with caret.

Reply:
I mean, waiting for the name to get switched seems like it would significantly delay things, right?  I'd prefer to pin simply so this can be merged quicker.

Reply:
My thinking is that we can resolve the name thing by next week, but others may not be so optimistic.

Anyway, to be more explicit: if we go with the experimental, semver-abandoning name, then yes we should pin.

Reply:
I vote for pinning for now, as Ganache has imminent releases queued up, and we're about to be OOO for a week.

Reply:
I do not think that this breaks semver for Ganache; this is clearly marked as a private member, as denoted by the `_`. And hopefully the extra `_` followed by "experimental" communicates that it is going to break. It is also undocumented.

If a property prefixed with `__experimental` does not sufficiently communicate that this one thing will change in the future then humanity has no hope of making it through The Great Filter; may God have mercy on our souls.

Reply:
Yeah, @davidmurdoch don't take my words too harshly; I just mean that the use experimental features means that we can't rely on semver, and thus pinning is logically more warranted.

Reply:
On last point: if removing something labeled `__experimental` without a major version bump is abandoning semver than fixing bugs is a much worse offender. :-)

Reply:
relevant: https://github.com/semver/semver/issues/238#issuecomment-1149260160

Reply:
> For this system to work, you first need to declare a public API. This may consist of documentation or be enforced by the code itself. Regardless, it is important that this API be clear and precise.

https://github.com/semver/semver/blob/master/semver.md

Per semver, it seems that documenting this endpoint as private (or at least "not public") in a clear and consider manner is all Ganache needs to do to be in absolute compliance. I still think__experimental_ is clear though. 🤷‍♂️

Reply:
So... I guess I pin for now, then?  Or what?

Reply:
> So... I guess I pin for now, then?  Or what?

Sounds like!

Reply:
OK, pin commit added.